crates/wasm-interpreter/src/lib.rs

```diff
@@ -68,11 +68,9 @@ impl Interpreter {
     pub fn new(module: &Module) -> Result<Interpreter, anyhow::Error> {
         let mut ret = Interpreter::default();
 
-        // The descriptor functions shouldn't really use all that much memory
-        // (the LLVM call stack, now the wasm stack). To handle that let's give
-        // our selves a little bit of memory and set the stack pointer (global
-        // 0) to the top.
-        ret.mem = vec![0; 0x400];
+        // Give ourselves some memory and set the stack pointer
+        // (the LLVM call stack, now the wasm stack, global 0) to the top.
+        ret.mem = vec![0; 0x8000];
```

Comment:
So this was... fun to debug

The wasm interpreter/runtime that's initialized here holds onto a stack pointer that starts at the highest memory location mem.len(), which was 0x400 originally (1024). Then, as the program runs and local variables are pushed onto the stack the stack pointer is decremented for each new variable. The other changes in this PR increased the complexity of some of the describes by a large amount, and so the stack pointer eventually got decremented so many times in one of those functions that it fell below 0 which is illegal (a.k.a stack overflow). So I increased the program memory up to 0x2000 (8192) and then the crash stopped happening.

In particular it was the GpuTextureFormat enum that was causing the problem due to the number of enum variants it has (and therefore the number of instructions required to implement the describe function).

This post helped me understand this issue: https://hackmd.io/@pepyakin/SkmPKGhiq